lib/model/message_list.dart

```diff
@@ -288,11 +316,14 @@ mixin _MessageSequence {
 
   /// Update [items] to include markers at start and end as appropriate.
   void _updateEndMarkers() {
+    assert(fetched);
     assert(!(haveOldest && fetchingOlder));
-    final startMarker = switch ((fetchingOlder, haveOldest)) {
-      (true, _) => const MessageListLoadingItem(MessageListDirection.older),
-      (_, true) => const MessageListHistoryStartItem(),
-      (_,    _) => null,
+    assert(!(fetchingOlder && fetchOlderCoolingDown));
+    final startMarker = switch ((fetchingOlder, haveOldest, fetchOlderCoolingDown)) {
+      (true, _, _) => const MessageListLoadingItem(MessageListDirection.older),
+      (_, true, _) => const MessageListHistoryStartItem(),
+      (_, _, true) => const MessageListLoadingItem(MessageListDirection.older),
```

Comment:
nit: the fact these two fetchOlder-related conditions produce the same result isn't a coincidence; so let's handle them together.

… Oh hmm but is the point that the order of these cases is significant? Is it possible to have `haveOldest` and `fetchOlderCoolingDown` both true?

If that is possible then I guess let's make that explicit. In the existing version in main, the `assert(!(haveOldest && fetchingOlder))` is there precisely to reassure the reader that the order of the existing two cases *doesn't* matter.

Reply:
I believe in fact it's the case in this revision that `haveOldest` and `fetchOlderCoolingDown` can't both be true. Which is good — it seems like that'd be a confused state to be in.

So let's (a) assert that, and (b) combine the two fetchOlder-related cases here. Could have a line like
```dart
  final effectiveFetchingOlder = fetchingOlder || fetchOlderCoolingDown;
```
and then the same `switch` cases as in main.